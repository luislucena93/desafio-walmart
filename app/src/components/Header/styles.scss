@import '../../commons/styleVars';

.headerBar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 0px 15%;
  height: 100px;
  background-color: $mainColor;
}

.logo {
  width: 200px;
  height: 80px;
}

.shoppingCart {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100px;
  background-color: $secondaryColor;
  padding: 10px;
  img {
    width: 50%;
  }
  div {
    border-radius: 50px;
    background-color: $yellow;
    height: 30px;
    width: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $mainColor;
    font-weight: 600;
  }
}