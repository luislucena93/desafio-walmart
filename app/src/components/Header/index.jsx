import React from 'react';
import './styles.scss';
import LogoImage from '../../assets/imgs/liderLogo.png';
import ShoppingCartImage from '../../assets/imgs/shopping-cart.png';

const Header = (props) => {
    const {selectedProducts, openShoppingCart} = props;
    let totalProducts = 0;
    Object.keys(selectedProducts).forEach((key) => {
        totalProducts += selectedProducts[key].count;
    });

    const handleOpenCart = () => {
        if (totalProducts !== 0) {
            openShoppingCart();
        }
    }

    return (
        <div className={"headerBar"}>
            <img alt="Logo" className={"logo"} src={LogoImage}/>
            <div className={"shoppingCart"} onClick={() => handleOpenCart()}>
                <img src={ShoppingCartImage} alt={"shoppingCartIcon"}/>
                <div data-testid={"totalProducts"}>{totalProducts}</div>
            </div>
        </div>
    )
};

export default Header;
