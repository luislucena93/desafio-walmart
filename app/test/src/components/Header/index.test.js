import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Header from "../../../../../app/src/components/Header/index";

describe('Header component', () => {

    let mockOpenShopingCart;
    const selectedProducts = {1: {count: 2, info: {}}};

    beforeEach(() => {
        mockOpenShopingCart = jest.fn();
    });

    it('should render header correctly for empty cart', () => {
        const {container} = render(<Header selectedProducts={{}} openShoppingCart={mockOpenShopingCart}/>);
        expect(container).toMatchSnapshot();
    });

    it('sould should render header correctly for cart with 2 products', () => {
        const {container} = render(<Header
            selectedProducts={selectedProducts} openShoppingCart={mockOpenShopingCart}/>);
        expect(container).toMatchSnapshot();
    });

    it('should call open shopping cart function when click in shopping cart icon if cart has products', () => {
        render(<Header selectedProducts={selectedProducts} openShoppingCart={mockOpenShopingCart}/>);
        fireEvent.click(screen.getByAltText('shoppingCartIcon'));
        expect(mockOpenShopingCart).toHaveBeenCalled();
    })

    it('should not call open shopping cart function when click in shopping cart icon if cart is empty', () => {
        render(<Header selectedProducts={{}} openShoppingCart={mockOpenShopingCart}/>);
        fireEvent.click(screen.getByAltText('shoppingCartIcon'));
        expect(mockOpenShopingCart).not.toHaveBeenCalled();
    })
});