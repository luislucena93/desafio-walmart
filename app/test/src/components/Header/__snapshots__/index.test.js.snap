// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render header correctly for empty cart 1`] = `
<div>
  <div
    class="headerBar"
  >
    <img
      alt="Logo"
      class="logo"
      src="test-file-stub"
    />
    <div
      class="shoppingCart"
    >
      <img
        alt="shoppingCartIcon"
        src="test-file-stub"
      />
      <div
        data-testid="totalProducts"
      >
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`Header component sould should render header correctly for cart with 2 products 1`] = `
<div>
  <div
    class="headerBar"
  >
    <img
      alt="Logo"
      class="logo"
      src="test-file-stub"
    />
    <div
      class="shoppingCart"
    >
      <img
        alt="shoppingCartIcon"
        src="test-file-stub"
      />
      <div
        data-testid="totalProducts"
      >
        2
      </div>
    </div>
  </div>
</div>
`;
